metaflow/metaflow_environment.py

```diff
@@ -76,21 +76,45 @@ def get_client_info(cls, flow_name, metadata):
         str : Information printed and returned to the user
         """
         return "Local environment"
+    
+    def get_s3_download_command(self, s3_path, local_path):
+        """
+
+        Parameters
+        ----------
+        s3_path : str
+            Path to S3 file
+        local_path : str
+            Path to download S3 file to
+        
+        Returns
+        -------
+        str : Command 
+        """
+
+        return (
+            "%s -c \"" % self._python()
+            + "import boto3; "
+            + "exec('try:\\n from urlparse import urlparse\\nexcept:\\n from urllib.parse import urlparse');"
+            + "parsed = urlparse('%s');" % s3_path
```

Comment:
@tkislan This parsing can happen while we are generating the `get_s3_download_command` locally rather than in the remote environment. That will save us precious entry point chars.